### PR TITLE
Convert replit module to V2

### PR DESCRIPTION
**Pull request by airportyh**
Why
===

Convert replit module to V2 scheme.

What changed
============

1. Moved dotreplit-lsp over
2. Added a replit bundle
3. Set the default enable value for replit bundle to true. This gives users the ability to opt-out by setting `bundles.replit.enable = false`

Test
===

1. `nix build .#v2.examples.replit`
2. `ls result`

### CODE REVIEW

File: pkgs/modules/languageServers/dotreplit-lsp/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{ pkgs, config, ... }:
+with pkgs.lib;
+let
+  cfg = config.languageServers.dotreplit-lsp;
+  # use changes from https://github.com/tamasfe/taplo/pull/510
```
Comment (airportyh) [inline pkgs/modules/languageServers/dotreplit-lsp/default.nix:5]: btw @cdmistman your stuff's been merged.

**Comment by airportyh** [comment]
Re: https://github.com/replit/nixmodules/pull/314#discussion_r1568988605 it's possible to default Python's `useSiteCustomize` to true when replit bundle is enabled, but I am not sure that's what we want, because a user wanting to disable the replit bundle doesn't mean they are not running in a repl.

File: pkgs/modules/languageServers/dotreplit-lsp/default.nix (hunk 1 above)
Comment (cdmistman) [reply pkgs/modules/languageServers/dotreplit-lsp/default.nix:5]: yup! was able to get it merged after pinging. still hasn't been a release yet though so once that's done we can go back to relying on their standard releases :)

**Comment by cdmistman** [comment]
> a user wanting to disable the replit bundle doesn't mean they are not running in a repl.

agreed, looks like we'll want some stdenv-style magic attrs that represent the nature of the user's environment

File: pkgs/modules/languageServers/dotreplit-lsp/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  # - use the above custom bin
+  taplo = pkgs.rustPlatform.buildRustPackage rec {
+    pname = "taplo";
+    version = "0.patched";
+    src = pkgs.fetchFromGitHub {
+      owner = "cdmistman";
+      repo = "taplo";
+      rev = "22eff1f7775e48eee8b50518c67f992b4595ab61";
+      hash = "sha256-63fm8pH03TJd4QBuhIxtttoEAaBnc9TuHGKCMK4YGP0=";
+    };
```
Comment (cdmistman) [inline pkgs/modules/languageServers/dotreplit-lsp/default.nix:27]: i'll update this soon to use their latest release + my patch, no action needed here though

**Review by cdmistman — APPROVED** [review]

**Pull request closed without merging**